### ISSUE TITLE
Nipy outlook

### ISSUE THREAD

**Issue by TheChymera** [nipy/nipy#413]
I heard somewhere (at brainhack?) that nipy is being discontinued. Is that true? If so, what happens to all the nice native-python preprocessing functions? What package takes the job over?

**Comment by matthew-brett** [comment]
See: https://github.com/nipy/nipy/pull/412#issuecomment-247203887

**Comment by TheChymera** [comment]
> The general plan is to keep nipy maintained for the next five years or so, but more active development of various parts has shifted to:
> 
> 1) registration : https://github.com/nipy/nireg (although it looks like development is a bit slow);
> 2) one of the two statistical interfaces : https://github.com/nistats/nistats
> 
> My provisional plan is to start development on something that might be called nipy2, with a changed interface, fixing some of the painful parts of the current nipy interface. At that point it would be a good idea to import from those modules, if they are in good shape.

I agree that "neuroiamging in python" might be a too-big job for one package. But my impression is that nilearn and nistats are (and should be, by virtue of their names) more focused on the data analysis (I feel nilearn is already amassing too much functionality for its name by providing all of that awesome plotting).

What I really don't understand is where all the preprocessing should end up. Should nireg just do non/linear/rigid/etc transformations (like pypreprocessing or dipy are *also* already providing)? Should it also do realignment, skullstripping, segmentation, spatial/temporal frequency filtering, or any of the other miscellaneous preprocessing tasks? Not least of all I can't dinf any documentation, examples, or tutorials for nireg.

**Comment by TheChymera** [comment]
[This](https://github.com/nipy/nireg/graphs/contributors) time course does not look to well. What do you think is the issue?

**Comment by matthew-brett** [comment]
Yes, the situation is bad at the moment.   We have had discussions in the past about trying to move things around in a coherent way, but they haven't worked out very well.   I think the particular problem with nireg is that Alexis is busy in his day job - but Alexis can say more I am sure.  I'm guessing he'd love some help organizing stuff and writing tutorials.

There were some discussions about moving the dipy registration stuff somewhere more obvious for general registration, such as nireg or scikit-image, but Eleftherios Garyfallidis, the main dipy developer, wanted to keep the registration code with the rest of the diffusion analysis stuff.

**Comment by TheChymera** [comment]
> but Eleftherios Garyfallidis, the main dipy developer, wanted to keep the registration code with the rest of the diffusion analysis stuff.

I've met him, he seems really nice, but I agree that that's an unfortunate decision, especially since I have been hearing good stuff about dipy's registration capabilities. Since it is BSD, though, I guess good bits from it can be copied into nireg, right? Of course, development will diverge and resources will continue to be split, but I guess competition is also good, especially when the projects can always take things from one another and keep some loose compatibility.

I am currently using ANTs for registration and I am sad to say I don't get the impression that the python environment is stable or mature enough to substitute it (afaik even dipy is missing elastic transform atm). But if help is needed organizing other preprocessing functions, I would be happy to contribute. Perhaps @alexis-roche can tell me a bit more about where this is going? But I do believe nireg might not be an obvious place to put realignment, skullstripping, spatial/temporal frequency filtering, etc.

**Comment by TheChymera** [comment]
I think by now it's pretty clear nipy isn't being discontinued :)
